### PR TITLE
Update when out of internal network & bug fixes

### PR DESCRIPTION
**Pull request by zhx828**
- Make sure the synonyms is always an array
- Do not allow changing preferred name
- Update tabs once the internal status is determined
- Use curation private for search drug
- Add a loading bar when changing the preferred name

### CODE REVIEW

File: web/yo/app/scripts/controllers/nav.js [hunk 1]
```diff
@@ -23,12 +23,13 @@ angular.module('oncokbApp')
             filterTabs.push('genes');
             filterTabs.push('queues');
             filterTabs.push('therapies');
-            if ($rootScope.me.admin) {
+            filterTabs.push('tools');
+            if ($rootScope.me && $rootScope.me.admin) {
                 filterTabs = _.union(filterTabs, ['variant', 'tools', 'feedback']);
```
Comment (victoria34) [inline web/yo/app/scripts/controllers/nav.js]: Are we supposed to remove `tools` in here?

File: web/yo/app/scripts/controllers/nav.js [hunk 2]
```diff
@@ -23,12 +23,13 @@ angular.module('oncokbApp')
             filterTabs.push('genes');
```
Comment (victoria34) [inline web/yo/app/scripts/controllers/nav.js]: Could we just write like this?
```
var filterTabs = ['genes', 'queues', 'therapies', 'tools'];  
```

File: web/yo/app/scripts/controllers/nav.js [hunk 3]
```diff
@@ -23,12 +23,13 @@ angular.module('oncokbApp')
...
             filterTabs.push('therapies');
-            if ($rootScope.me.admin) {
+            filterTabs.push('tools');
+            if ($rootScope.me && $rootScope.me.admin) {
                 filterTabs = _.union(filterTabs, ['variant', 'tools', 'feedback']);
             }
 
             if (!$rootScope.internal) {
-                filterTabs = _.intersection(filterTabs, ['genes', 'queues', 'feedback']);
+                filterTabs = _.intersection(filterTabs, ['genes', 'queues', 'feedback', 'therapies', 'tools']);
```
Comment (victoria34) [inline web/yo/app/scripts/controllers/nav.js]: Does it mean we can only see the `Variant Annotation` tab when we are in external network?

File: web/yo/app/scripts/controllers/nav.js (hunk 1 above)
Comment (zhx828) [reply web/yo/app/scripts/controllers/nav.js]: Tools is only available for admins

File: web/yo/app/scripts/controllers/nav.js (hunk 2 above)
Comment (zhx828) [reply web/yo/app/scripts/controllers/nav.js]: This is previous setting. I think it's ok.

File: web/yo/app/scripts/controllers/nav.js (hunk 3 above)
Comment (zhx828) [reply web/yo/app/scripts/controllers/nav.js]: This is intersection when user is in external. Variant Annotation should only available internal

**Comment by Yizhuzhu** [comment]
I downloaded it and did a test. I have two concerns.
1. Change network from internal to public website. Nothing changed in nav bar and tools page.
2. Changing drug preferred name. When I clicked the check button to save, nothing responded during two seconds and then the page flashed. I guess it was loading page. Some words on the page, but it flashed too quickly. It's hard to read it.

**Comment by zhx828** [comment]
@Yizhuzhu for your concerns:
1: I don't think in general we have the solution when toggle from internal to external or the other way. User has to do refresh the page.
2: The reason I added the loading status is because changing preferred name is very expensive. It usually takes about one minute or even longer. The server side needs to do full recaching. So I don't think it will flash. But could you explain a bit more about `nothing responded`?

File: web/yo/app/scripts/controllers/nav.js (hunk 1 above)
Comment (victoria34) [reply web/yo/app/scripts/controllers/nav.js]: So we should remove line 26 right?

**Comment by Yizhuzhu** [comment]
@zhx828 After clicking save button, I need to wait about 2 seconds, then the page flashes in about 1 second. Then it goes back to drug page. The time may not be exact. However, flashing time is less than waiting time.

File: web/yo/app/scripts/controllers/nav.js (hunk 1 above)
Comment (zhx828) [reply web/yo/app/scripts/controllers/nav.js]: Ah yes, let me do that.

**Comment by zhx828** [comment]
@Yizhuzhu there is no so called flashing time is less than waiting time. The loading bar will be shown for quite long(see my comment above). I update the loading status right after the saveDrugName is called. It may just due to the lifecycle. No sure what I can do here. But I'm happy to invest in more time on this issue later. But I don't think it's a priority at the moment.

**Pull request merged**